src/compas_timber/elements/fasteners/fastener.py

```diff
@@ -0,0 +1,165 @@
+from compas_model.elements import Element
+from compas_model.elements import reset_computed
+
+
+class Fastener(Element):
+    """
+    A class to represent timber fasteners (screws, dowels, brackets).
+
+    Parameters
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    Attributes
+    ----------
+    elements : list(:class:`~compas_timber.parts.Element`)
+        The elements that are connected with this fastener.
+
+    """
+
+    def __init__(self, elements, **kwargs):
+        super(Fastener, self).__init__(elements, **kwargs)
+        self.elements = elements
```

Comment:
why are the elements saved here? are these the elements (beams) this fastener connects?
if so then it's up to the model to describe and maintain this relationship.